vividus-http-client/src/main/java/org/vividus/http/handler/ValidatingStatusCodeHandler.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2019-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.http.handler;
+
+import static org.apache.commons.lang3.Validate.isTrue;
+
+import java.io.IOException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.vividus.http.client.HttpResponse;
+
+public class ValidatingStatusCodeHandler implements HttpResponseHandler
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(ValidatingStatusCodeHandler.class);
+
+    private final int startStatusCode;
+    private final int endStatusCode;
+
+    @SuppressWarnings("MagicNumber")
+    public ValidatingStatusCodeHandler(int startStatusCode, int endStatusCode)
+    {
+        isTrue(startStatusCode < endStatusCode, "Left status code must be lower than right status code");
+        isTrue(startStatusCode >= 100 && endStatusCode <= 599, "The status codes must fit into 100-599 range");
+        this.startStatusCode = startStatusCode;
+        this.endStatusCode = endStatusCode;
+    }
+
+    @Override
+    public void handle(HttpResponse httpResponse) throws IOException
+    {
+        int status = httpResponse.getStatusCode();
+        if (status >= startStatusCode && status < endStatusCode)
+        {
+            return;
+        }
+        LOGGER.atError().addArgument(httpResponse).log("Response: {}");
```

Comment:
The problem here is that we don't know "who" is actual client, I guess we need to init logger with external class

Reply:
bad idea, it's better to put this to message, like `"Service response: {}"`, e.g. `"JIRA response: {}"`

Reply:
on the other hand it might make debugging process more complex